### PR TITLE
NAS-130116 / 24.10 / Fix ipa join admin privileges and keytab

### PR DESCRIPTION
**Pull request by anodos325**
This adds back in missing cherry-pick from development branch to properly grant privileges to the admins group in an IPA domain.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/ldap.py [hunk 1]
```diff
@@ -918,22 +921,21 @@ async def __start(self, job, ds_type):
                 ds_type = DSType.LDAP
                 await self.middleware.call('directoryservices.health.set_state', ds_type.value, DSStatus.JOINING.name)
 
-        cache_job_id = await self.middleware.call('directoryservices.connection.activate')
+        if dom_join_resp == DomainJoinResponse.ALREADY_JOINED.name:
+            cache_job_id = await self.middleware.call('directoryservices.connection.activate')
+            await job.wrap(await self.middleware.call('core.job_wait', cache_job_id))
 
-        await job.wrap(await self.middleware.call('core.job_wait', cache_job_id))
         await self.middleware.call('directoryservices.health.set_state', ds_type.value, DSStatus.HEALTHY.name)
```
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/ldap.py:928]: The `set_state` to `HEALTHY` here doesn't seem correct.  We _could_ have just set it to JOINING above.

_Maybe_ indent under
```
if dom_join_resp in (DomainJoinResponse.ALREADY_JOINED.value, DomainJoinResponse.PERFORMED_JOIN.value):
```
line?

(Actually now I'm wondering whether we need this same condition above too.)

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/ldap.py [hunk 2]
```diff
@@ -897,13 +897,16 @@ async def __start(self, job, ds_type):
         ldap = await self.config()
 
         await self.middleware.call('ldap.create_sssd_dirs')
+        dom_join_resp = DomainJoinResponse.ALREADY_JOINED.name
```
[the remark this answers is not in the document]
Comment (anodos325) [reply src/middlewared/middlewared/plugins/ldap.py:900]: Derp. Yup.

File: src/middlewared/middlewared/plugins/ldap.py (hunk 1 above)
Comment (anodos325) [reply src/middlewared/middlewared/plugins/ldap.py:928]: I'll add comments.

**Review by mgrimesix — APPROVED** [review]
My changes were corrected, might wait for Brian's approval.

File: tests/api2/test_ipa_join.py [hunk 3]
```diff
@@ -85,7 +85,9 @@ def test_smb_keytab_exists(do_freeipa_connection):
 def test_admin_privilege(do_freeipa_connection, enable_ds_auth):
     ipa_config = call('ldap.ipa_config')
 
-    priv = call('privilege.query', [['name', '=', ipa_config['domain']]], {'get': True})
+    priv_names = [priv['name'] for priv in call('privilege.query')]
```
Comment (bmeagherix) [inline tests/api2/test_ipa_join.py:88]: priv is now undefined below

Also 2 flake8 indentation is not a multiple of 4 on lines 21 & 30 above

**Review by bmeagherix — CHANGES_REQUESTED** [review]

File: src/middlewared/middlewared/plugins/ldap.py [hunk 4]
```diff
@@ -918,22 +921,39 @@ async def __start(self, job, ds_type):
...
+            case DomainJoinResponse.ALREADY_JOINED.value:
+                cache_job_id = await self.middleware.call('directoryservices.connection.activate')
+                await job.wrap(await self.middleware.call('core.job_wait', cache_job_id))
+                # Change state to HEALTHY before performing final health check
+                await self.middleware.call('directoryservices.health.set_state', ds_type.value, DSStatus.HEALTHY.name)
+                # Force health check so that user gets immediate feedback if something
+                # went sideways while enabling
+                await self.middleware.call('directoryservices.health.check')
+            case _:
+                raise CallError(f'{dom_join_response}: unexpected domain join response')
```
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/ldap.py:945]: Typo: `dom_join_response` should be `dom_join_resp`

**Review by bmeagherix — APPROVED** [review]
Approve, but one fix still required.

**Pull request merged**